### ISSUE TITLE
[REQ][JAVA][WebClient] Add @Component ?

### ISSUE THREAD

**Issue by UnleashSpirit** [OpenAPITools/openapi-generator#10023]
Hi, it's more a question cause I have no real idea.

Do we "need" to add the `@Component`  Spring annotation over generated Api in WebClient like in RestTemplate ?
```java
@javax.annotation.Generated(...)
@Component  <== HERE ?
public class PetApi {
    [...]
}
```

I don't know if it can cause any side effects ?

**Comment by UnleashSpirit** [comment]
The goal os this question is to avoid the @Bean creation of the PetApi

**Comment by matjamesymj** [comment]
@UnleashSpirit  any update on this. You used to get @Component on generated clients using RestTemplate. You also got @JsonProperty on model fields with resttemplate. It's stopping us from moving from RestTemmplate to webclient at the moment.

**Comment by UnleashSpirit** [comment]
Hi, till now we just declare a bean for each api in a @Service config class
But you can also use the override template mechanism if you prefer

Fix this I "easy" I think but I don't know if any side effect are possible, hence my question

**Comment by matjamesymj** [comment]
Yeah that's what we are doing now. Also ignore my comment on @JsonProperty missing. Thanks for the reply
